src/test/java/com/google/devtools/build/lib/bazel/repository/cache/RepositoryCacheTest.java

```diff
@@ -171,6 +174,18 @@ public void testGetChecksum() throws Exception {
     assertThat(actualChecksum).isEqualTo(downloadedFileSha256);
   }
 
+  @Test
+  public void testGetChecksumWithFastDigest() throws Exception {
+    String fastDigestChecksum = "cfe5ed57e6e323555b379c660aa8d35b70c2f8f07cf03ad6747266495ac13be0";
+
+    FileStatus fileStatus = downloadedFile.stat();
+    assertThat(fileStatus).isInstanceOf(InMemoryFileInfo.class);
+    ((InMemoryFileInfo) fileStatus).setFastDigest(BaseEncoding.base16().lowerCase().decode(fastDigestChecksum));
```

Comment:
this looks very suspect. `fastDigestChecsum` looks like a normal hexadecimal string, and yet you're decoding it as if it's base64. This might accidentally work since any hexadecimal string of length 4n is a valid base64 encoded string, but semantically is very much not what you want.

Reply:
Isn't `BaseEncoding.base16()` base16, not hex?

Reply:
same as above, you're all good :)